### PR TITLE
fix: unescape \\ for levelKey

### LINKED ISSUE #456: Regression with levelKey handling in version 10
When pino-pretty is used together with elastic formatting the logging level has `log.level` key in the log object. I found no way to make pino-pretty work correctly with this level key.

In version 9 pino-pretty allowed dot in key names without escaping while in version 10 we need to escape it with slashes.
This would be ok but this escaping breaks excluding this key from the message string.

Example (uncomment levelKey with/without escaping to see different havaviours):

```javascript
const pino = require('pino');
const pretty = require('pino-pretty');
const ecsFormat = require('@elastic/ecs-pino-format');

const options = ecsFormat({});
options.level = 'debug';
const logger = pino(options, pretty({
    colorize: true,
    singleLine: true,
    levelKey: 'log\\.level', // if levelKey dot is escaped then log.level appears at the console output
    // levelKey: 'log.level', // if levelKey dot is not escaped then level is not showing between message
    timestampKey: '@timestamp',
    messageKey: 'message',
    ignore: 'service,process,host,ecs'
}));

logger.info('Starting');
logger.debug({
    extraParam: '123',
    message: 'hello hello'
});
```

The example output is 
image

As you can see the output contains unwanted `"log.level":"info"` and `"log.level":"debug"`.

If I remove dot escaping then the output changes to
image

Ths time it has no colored `INFO` and `DEBUG` levels.

### PR DESCRIPTION
**Pull request by eliw00d**
closes #456

### CODE REVIEW

File: lib/pretty.js [hunk 1]
```diff
@@ -147,11 +147,13 @@ function pretty (inputData) {
       this.messageKey,
       this.levelKey,
       this.timestampKey
-    ].filter(key => {
-      return typeof log[key] === 'string' ||
-        typeof log[key] === 'number' ||
-        typeof log[key] === 'boolean'
-    })
+    ]
+      .map((key) => key.replaceAll(/\\/g, ''))
```
Comment (eliw00d) [inline lib/pretty.js:151]: This strips off the `\\` in `log\\.level` so that you get the correct key to ignore.

File: test/basic.test.js [hunk 2]
```diff
@@ -218,6 +218,22 @@ test('basic prettifier tests', (t) => {
...
 
+  t.test('can use nested level keys', (t) => {
+    t.plan(1)
+    const pretty = prettyFactory({ levelKey: 'log\\.level' })
+    const log = pino({}, new Writable({
+      write (chunk, enc, cb) {
+        const formatted = pretty(chunk.toString())
+        t.equal(
+          formatted,
+          `[${formattedEpoch}] WARN (${pid}): foo\n`
```
Comment (eliw00d) [inline test/basic.test.js:229]: Without this fix, you would get `log.level`  underneath in the prettified object.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by jsumners** [comment]
Yes, we should drop 14. It'll be a new major, though.

**Comment by eliw00d** [comment]
> Yes, we should drop 14. It'll be a new major, though.

Should I make that change in this PR? For example:

```yml
      matrix:
        node-version: [18.17, 20, 22]
        os: [ubuntu-latest]
        pino-version: [^9.0.0]
```

**Comment by mcollina** [comment]
@eliw00d send a fresh PR.

**Comment by eliw00d** [comment]
> @eliw00d send a fresh PR.

[Done](https://github.com/pinojs/pino-pretty/pull/540).

**Comment by eliw00d** [comment]
@jsumners @mcollina Should be good to go now! 🎉

**Review by mcollina — APPROVED** [review]
lgtm

**Review by jsumners — APPROVED** [review]

**Pull request merged**